src/main/scala/dsptools/counters/Counters.scala

```diff
@@ -164,11 +165,13 @@ abstract class Counter(countParams: CountParams) extends Module {
       //    else Mux(wrap,wrapTo,nextInSeq)
     }
     else {
-      if(wrap == Bool(true)) {
-        wrapTo
-      } else {
-        nextInSeq
+      val nextThing = Wire(Bool())
```

Comment:
[My] Original code was 
```
val nextCount = {
    if (countParams.wrapCtrl == Internal && (countParams.countType == UpMod || 
       (countParams.countType == Up && countParams.incMax > 1 && !countParams.customWrap))) 
      nextInSeq
    else Mux(wrap,wrapTo,nextInSeq)
  }
```

Which is compile time-configurable.

Anyways, didn't code review the changes made. My original counter, as verbose as it was, should have been functionally correct and runtime/compile-time configurable.